### PR TITLE
Retrieve binary secrets from AWS Secrets Manager

### PR DESCRIPTION
**Pull request by arruzk**
Should fix https://github.com/godaddy/kubernetes-external-secrets/issues/149 issue

### CODE REVIEW

**Comment by arruzk** [comment]
@silasbw Flydiverny Could you please take a look?

File: lib/backends/secrets-manager-backend.js [hunk 1]
```diff
@@ -39,6 +39,9 @@ class SecretsManagerBackend extends KVBackend {
       .getSecretValue({ SecretId: secretKey })
       .promise()
 
+    if (data.SecretString === undefined) {
```
Comment (silasbw) [inline lib/backends/secrets-manager-backend.js]: can you tweak this to explicitly check for the existence of the properties?  I.e.,

```js
if ('SecretBinary' in data) {
  return data.SecretBinary
} else if ('SecretString' in data) {
  return data.SecretString
}

this._logger('Unexpected data from Secrets Manager secret ${secretKey}')
return null
```

Comment (arruzk) [reply lib/backends/secrets-manager-backend.js]: @silasbw thank you for the comment, I fixed it.

File: lib/backends/secrets-manager-backend.js [hunk 2]
```diff
@@ -39,7 +39,14 @@ class SecretsManagerBackend extends KVBackend {
...
       .promise()
 
-    return data.SecretString
+    if ('SecretBinary' in data) {
+      return data.SecretBinary
+    } else if ('SecretString' in data) {
+      return data.SecretString
+    }
+
+    this._logger(`Unexpected data from Secrets Manager secret ${secretKey}`)
```
Comment (Flydiverny) [inline lib/backends/secrets-manager-backend.js]:
```suggestion
    this._logger.error(`Unexpected data from Secrets Manager secret ${secretKey}`)
```

Comment (arruzk) [reply lib/backends/secrets-manager-backend.js]: Thank you! I changed logging level

**Pull request merged**